bin/puppet-validator

```diff
@@ -89,5 +102,24 @@ else
   logger.level     = loglevel
   options[:logger] = logger
 
-  PuppetValidator.run! options
+  if ssl_opts[:cert_chain_file] and ssl_opts[:private_key_file]
+    options[:ssl] = true
+  end
+
+  # These options should either both be nil or both be Strings
+  unless ssl_opts[:cert_chain_file].class == ssl_opts[:private_key_file].class
```

Comment:
This may be Ruby ignorance on my part here, but do you also have to check that the value of `options[:ssl]` is `true` here to make sure this expression is only checked then? And should this expression also make sure `.class` results in `String`?

Reply:
Nope, though it could potentially be explained a bit more. The values start out as unset (which evaluates as `nil`) and then possibly get set by getopts, which can only specify strings unless you coerce it yourself. So the only two possible values are `nil` and `String`, so we just check to make sure that they're both the same type. If you set one, you must set both.